skimage/measure/_regionprops_utils.py

```diff
@@ -326,3 +327,28 @@ def perimeter_crofton(image, directions=4):
 
     total_perimeter = coefs @ h
     return total_perimeter
+
+def check_spacing(spacing, ndims:int):
+    """Checks the spacing parameter
+
+    The `spacing` parameter should be a tuple.
+    If the spacing is a single number (e.g. float)
+    then assume spacing is same in all dimensions.
+
+    Parameters
+    ---------
+    spacing: The spacing of the pixels in the image
+
+    Returns
+    -------
+    Corrected spacing as tuple (if needed)
+    """
+
+    if isinstance(spacing, (tuple, List)):
+        if not np.all([np.isreal(x) for x in spacing]):
+            raise ValueError(f'Spacing should be a tuple of {ndims} floats.')
+        return spacing
+    elif isinstance(spacing, (float, int)):
```

Comment:
Maybe isreal here too, since we don't know exactly what scalar object will be passed in. @seberg is isreal idiomatic, or is there any better way to check for "some kind of real number".

Reply:
I think `isinstance(<numpy scalar>, (np.integer, np.floating))` as recommended by [NumPy's scalar guide](https://numpy.org/devdocs/reference/arrays.scalars.html#scalars) could work? Though it's a bit awkward that it doesn't seem to work for native Python types. E.g. `isinstance(3, (np.integer, np.floating))` returns `False` so it requires calling `np.array(spacing)` first... :thinking:

Reply:
```
>>> isinstance(4., (np.integer, np.floating))
False
>>> isinstance(4, (np.integer, np.floating))
False
>>> isinstance(np.array(4), (np.integer, np.floating))
False
>>> isinstance(np.array(4.), (np.integer, np.floating))
False
```

Is this what you mean? It's returning false. Maybe I am misunderstanding your recommendation.

Reply:
```
from numbers import Number

elif isinstance(spacing, Numbers):
        return np.full(ndims, float(spacing))
```
What about this?

I found it in the numpy docs: https://numpy.org/doc/stable/reference/generated/numpy.isscalar.html

> If you need a stricter way to identify a numerical scalar, use isinstance(x, numbers.Number), as that returns False for most non-numerical elements such as strings.

Reply:
I suspect you mean `from numbers import Real` since Number would allow complex.  Unless you want to accept 0-D arrays or so, that seems good to me.  `isreal` is a bit weird/different, since it returns `np.imag(x) == 0`.

Reply:
Thanks. Corrected.

Reply:
I meant it this way
```python
spacing = np.array([2.5, 3.0])
isinstance(spacing[0], (np.integer, np.floating))  # True
```
But I think @seberg's suggestion is the better one!